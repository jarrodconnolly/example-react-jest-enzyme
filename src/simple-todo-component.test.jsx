import SimpleToDoComponent from './simple-todo-component';

const mockToDoData = {
  todos: [
    {name: 'Unit Test Item 1', done: true},
    {name: 'Unit Test Item 2', done: false},
    {name: 'Unit Test Item 3', done: false},
    {name: 'Unit Test Item 4', done: true},
  ]
};

const mockFetchToDoData = jest.fn();
SimpleToDoComponent.prototype.fetchDataFromApi = mockFetchToDoData;

beforeEach(() => {
  mockFetchToDoData.mockReset();
});

describe('Simple ToDo Component', () => {
  it('should render correctly', (done) => {
    const wrapper = shallow(<SimpleToDoComponent />);
    wrapper.setState(mockToDoData);

    process.nextTick(() => {
      expect(wrapper).toMatchSnapshot();
      expect(mockFetchToDoData.mock.calls.length).toBe(1);
      done();
    });
  });
});
