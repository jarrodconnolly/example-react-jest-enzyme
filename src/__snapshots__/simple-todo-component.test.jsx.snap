// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple ToDo Component should render correctly 1`] = `
<div>
  <h2>
    ToDo
  </h2>
  <ul>
    <li
      key="0"
    >
      Unit Test Item 1
       - 
      ✓
    </li>
    <li
      key="1"
    >
      Unit Test Item 2
       - 
      ✗
    </li>
    <li
      key="2"
    >
      Unit Test Item 3
       - 
      ✗
    </li>
    <li
      key="3"
    >
      Unit Test Item 4
       - 
      ✓
    </li>
  </ul>
</div>
`;
